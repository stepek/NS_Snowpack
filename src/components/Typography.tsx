import React, {ReactText} from "react"

type Variant = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "body" | "body1"

interface TypographyProps {
  variant?: Variant
  children?: ReactText
}
export function Typography({variant, children = ""}: TypographyProps) {
  switch (variant) {
    case "h1":
    case "h2":
    case "h3":
    case "h4":
    case "h5":
    case "h6":
    case "body":
    case "body1":
    default:
      return <span>{children}</span>
  }
}
export default Typography
