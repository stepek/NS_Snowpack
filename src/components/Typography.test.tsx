import React from "react"
import renderer from "react-test-renderer"

import Typography from "./Typography"

test("<Typography> render correctly default variant", () => {
  expect(renderer.create(<Typography />).toJSON()).toMatchSnapshot()
})
