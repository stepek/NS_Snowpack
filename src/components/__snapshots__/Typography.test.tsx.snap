// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography> render correctly default variant 1`] = `
<span>
  
</span>
`;
